mzLib/UsefulProteomicsDatabases/ProteinDbLoader.cs

```diff
@@ -430,15 +526,119 @@ public static IEnumerable<Protein> Merge(IEnumerable<Protein> mergeThese)
                     disulfideBonds: bonds.ToList(),
                     sequenceVariations: variants.ToList(),
                     spliceSites: splices.ToList(),
+                    databaseFilePath: dbFilePath, // keep original source path
                     dataset: datasets.FirstOrDefault(),
                     created: createds.FirstOrDefault(),
                     modified: modifieds.FirstOrDefault(),
                     version: versions.FirstOrDefault(),
                     xmlns: xmlnses.FirstOrDefault()
-                    );
+                );
             }
         }
+        /// <summary>
+        /// Finds groups of proteins that share the same accession and base sequence.
+        /// Intended to identify cases where an applied-variant entry appears twice
+        /// (e.g., one parsed from XML and another created via variant expansion).
+        /// </summary>
+        internal static IEnumerable<IGrouping<(string accession, string baseSequence), Protein>> FindDuplicateGroupsByAccessionAndBaseSequence(
```

Comment:
Why create a second set of merging duplicate functions when we already have a merge function? 

Is this functionality essential to the proper loading of sequence variants? If so, RNA will need it as well.